Library/Homebrew/tab.rb

```diff
@@ -70,29 +85,70 @@ def self.create(formula, compiler, stdlib)
           "version_scheme" => formula.version_scheme,
         },
       },
-      "built_on"                => DevelopmentTools.build_system_info,
-    }
+    })
 
     # We can only get `tap_git_head` if the tap is installed locally
     attributes["source"]["tap_git_head"] = formula.tap.git_head if formula.tap&.installed?
 
     new(attributes)
   end
 
-  # Returns the {Tab} for an install receipt at `path`.
+  def self.create_from_cask(cask)
+    attributes = generic_attributes(cask).merge({
+      "tabfile"              => cask.metadata_main_container_path/FILENAME,
+      "caskfile_only"        => cask.caskfile_only?,
+      "runtime_dependencies" => Tab.cask_runtime_deps_hash(cask, cask.depends_on),
+      "source"               => {
+        "path"         => cask.sourcefile_path.to_s,
```

Comment:
```suggestion
        "path"         => cask.sourcefile_path&.to_s,
```

As it can be nil here.

Reply:
I think we should avoid this being nil and set it to a sensible value (for formulae and casks).

Reply:
@apainintheneck can you advise on the best way to set this? I mean we can handle the logic within `Tab` to just add the core path if it's blank, the tap is homebrew/core, and we loaded from the api. But I feel like the better solution is for `caskfile_path` to be just return this value. You seemed more familiar with why this was blank in the first place